services:
  db:
    image: docker.io/library/postgres:14
    restart: always
    environment:
      - POSTGRES_USER=gitea
      - POSTGRES_PASSWORD=gitea
      - POSTGRES_DB=gitea
    networks:
      - gitea
    volumes:
      - ./gitea/data/postgres:/var/lib/postgresql/data
  server:
    image: docker.gitea.com/gitea:latest
    container_name: gitea
    environment:
      - USER_UID=1000
      - USER_GID=1000
      - GITEA__database__DB_TYPE=postgres
      - GITEA__database__HOST=db:5432
      - GITEA__database__NAME=gitea
      - GITEA__database__USER=gitea
      - GITEA__database__PASSWD=gitea
    restart: always
    networks:
      - gitea
    volumes:
      - ./gitea/data/server:/data
      - ./gitea/config/app.ini:/data/gitea/conf/app.ini
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro
    ports:
      - "3000:3000"
      - "222:22"
    depends_on:
      - db
  runner:
    image: docker.io/gitea/act_runner:latest
    environment:
      CONFIG_FILE: /config.yaml
      GITEA_INSTANCE_URL: "http://gitea:3000"
      GITEA_RUNNER_REGISTRATION_TOKEN: "49xy3Qbm4SGw6XlOicQCZL90fGf1OeFJb0rhrOrY"
      GITEA_RUNNER_NAME: "Gitea Runner"
    networks:
      - gitea
    volumes:
      - ./gitea/config/config.yaml:/config.yaml
      #- ./gitea/data/runner:/data
      - /var/run/docker.sock:/var/run/docker.sock
    depends_on:
      - server

networks:
  gitea:
    external: false
